src/lib/homestore_backend/hs_blob_manager.cpp

```diff
@@ -1,17 +1,220 @@
 #include "hs_homeobject.hpp"
+#include "replication_message.hpp"
+#include "replication_state_machine.hpp"
+#include "lib/homeobject_impl.hpp"
+
+SISL_LOGGING_DECL(blobmgr)
 
 namespace homeobject {
 
-BlobManager::Result< blob_id_t > HSHomeObject::_put_blob(ShardInfo const&, Blob&&) {
-    return folly::makeUnexpected(BlobError::UNKNOWN);
+BlobManager::Result< blob_id_t > HSHomeObject::_put_blob(ShardInfo const& shard, Blob&& blob) {
+    auto pg_id = shard.placement_group;
+    HS_PG* pg{nullptr};
+    {
+        std::shared_lock lg{_pg_lock};
+        if (auto it = _pg_map.find(pg_id); it != _pg_map.end()) {
+            pg = s_cast< HS_PG* >(it->second.get());
+        } else {
+            LOGWARN("failed to put blob with non-exist pg [{}]", pg_id);
+            return folly::makeUnexpected(BlobError::UNKNOWN_PG);
+        }
+    }
+
+    if (!pg->repl_dev_) {
```

Comment:
What is the scenario this can happen? Should we ASSERT instead?

Reply:
We get api requests while we are booting and pg recovery happened but not repl dev ?

Reply:
No, you cannot get a `shared_ptr<BlobManager>` or any other manager until `init` completes. So we can make this an assertion.

Reply:
Agree